core/app/models/spree/promotion.rb

```diff
@@ -27,35 +27,35 @@ class Promotion < Spree::Base
 
     before_save :normalize_blank_values
 
-    scope :coupons, ->{ where("#{table_name}.code IS NOT NULL") }
-    scope :applied, lambda {
-      joins(<<-SQL).uniq
-        INNER JOIN spree_order_promotions
-        ON spree_order_promotions.promotion_id = #{table_name}.id
-      SQL
-    }
+    scope :coupons, -> { where.not(code: nil) }
+    scope :advertised, -> { where(advertise: true) }
+    scope :active, -> do
+      table = self.arel_table
+      time = Time.current
+      where(table[:starts_at].not_eq(nil).and(table[:expires_at].not_eq(nil))).
```

Comment:
Correct me if I'm wrong, but shouldn't this scope look like:
```
where(table[:starts_at].not_eq(nil).and(table[:starts_at].lt(time))).where(table[:expires_at].not_eq(nil).and(table[:expires_at].gt(time)))
```
?

Because first ```where``` filters out Promotions with ```starts_at``` and ```expires_at``` == nil.

From now it could be even shorten to:
```
where(table[:starts_at].lt(time).and(table[:expires_at].gt(time)))
```
cause it automaticaly filters out nils.